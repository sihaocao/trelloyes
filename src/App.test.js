// import React from 'react';
// import ReactDOM from 'react-dom';
// import App from './App';

// it('renders without crashing', () => {
//   const div = document.createElement('div');
//   ReactDOM.render(<App />, div);
//   ReactDOM.unmountComponentAtNode(div);
// });



import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import Card from './card';
import List from './list';

describe('Card component', () => {
    it('renders without crashing', () => {
        const div = document.createElement('div');
        ReactDOM.render(<Card />, div);
        ReactDOM.unmountComponentAtNode(div);
    });

    it('renders the UI as expected', () => {
        const tree = renderer
            .create(<Card title='First Card' content='Something...' />)
            .toJSON();
        expect(tree).toMatchSnapshot();
    });


    it('renders without crashing', () => {
        const div = document.createElement('div');
        const props = {
            cards: [
                {
                    id: 'a',
                    title: 'First card',
                    content: 'lorem ipsum'
                }
            ]
        }
        ReactDOM.render(<List {...props} />, div);
        ReactDOM.unmountComponentAtNode(div);
    });

    it('renders the UI as expected', () => {
        const props = {
            cards: [
                {
                    id: 'a',
                    title: 'First card',
                    content: 'lorem ipsum'
                }
            ]
        }

        const tree = renderer
          .create(<List header='First List' {...props} />)
          .toJSON();
        expect(tree).toMatchSnapshot();
    });
});