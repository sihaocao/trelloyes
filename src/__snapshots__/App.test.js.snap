// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card component renders the UI as expected 1`] = `
<div
  className="Card"
>
  <button
    type="button"
  >
    delete
  </button>
  <h3>
    First Card
  </h3>
  <p>
    Something...
  </p>
</div>
`;

exports[`Card component renders the UI as expected 2`] = `
<section
  className="List"
>
  <header
    className="List-header"
  >
    <h2>
      First List
    </h2>
  </header>
  <div
    className="List-cards"
  >
    <div
      className="Card"
    >
      <button
        type="button"
      >
        delete
      </button>
      <h3>
        First card
      </h3>
      <p>
        lorem ipsum
      </p>
    </div>
    <button
      className="List-add-button"
      type="button"
    >
      + Add Random Card
    </button>
  </div>
</section>
`;
